tutorials/physics/rigid_body.rst

```diff
@@ -29,49 +29,52 @@ The "look at" method
 --------------------
 
 As described above, using the Node3D's ``look_at()`` method can't be used each frame to follow a target.
-Here is a custom ``look_at()`` method that will work reliably with rigid bodies:
+Here is a custom ``look_at()`` method that will work with rigid bodies:
 
 .. tabs::
  .. code-tab:: gdscript GDScript
 
     extends RigidBody3D
 
-    func look_follow(state, current_transform, target_position):
-        var up_dir = Vector3(0, 1, 0)
-        var cur_dir = current_transform.basis * Vector3(0, 0, 1)
-        var target_dir = current_transform.origin.direction_to(target_position)
-        var rotation_angle = acos(cur_dir.x) - acos(target_dir.x)
-
-        state.angular_velocity = up_dir * (rotation_angle / state.step)
+    var speed: float = 0.1
 
+    func look_at(state: PhysicsDirectBodyState3D, current_transform: Transform3D, target_position: Vector3) -> void:
+        var forward_local_axis: Vector3 = Vector3(1, 0, 0)
+        var forward_dir: Vector3 = (current_transform.basis * forward_local_axis).normalized()
+        var target_dir: Vector3  = (target_position - current_transform.origin).normalized()
+        var local_speed: float = clampf(speed, 0., acos(forward_dir.dot(target_dir)))
+        if (forward_dir.dot(target_dir) > 1e-4):
+            state.angular_velocity = local_speed*(forward_dir.cross(target_dir))/state.step
+    
     func _integrate_forces(state):
         var target_position = $my_target_node3d_node.global_transform.origin
-        look_follow(state, global_transform, target_position)
+        look_at(state, global_transform, target_position)
 
  .. code-tab:: csharp
 
     using Godot;
+    using System;
 
-    public partial class MyRigidBody3D : RigidBody3D
+    public partial class RigidBody3D : Godot.RigidBody3D
```

Comment:
```suggestion
    public partial class MyRigidBody3D : RigidBody3D
```
This is incorrect, we always name `My` and you're using the same name as a built in class